client/src/components/TaskCard/TaskCard.js

```diff
@@ -8,17 +8,27 @@ const TaskCard = ({
   location,
   startTime,
   endTime,
-  isOngoing,
+  ongoing,
+  finished,
   showDetailsDialog,
   edittable,
 }) => {
-  const [taskOngoing, setOngoing] = useState(isOngoing);
-  const taskTextStyle = !taskOngoing ? styles.taskDone : "";
+  const [taskFinished, setTaskFinished] = useState(finished);
+  const [taskOngoing, setTaskOngoing] = useState(ongoing);
+  const taskTextStyle = taskFinished ? styles.taskDone : "";
   const actionBtn = edittable ? (
     <SmallActionButton
       text={!taskOngoing ? "Start" : "End"}
       toggleButton={() => {
-        setOngoing(!taskOngoing);
+        if (taskOngoing) {
```

Comment:
Is this kept here as a placeholder?

Reply:
Ideally, we should be passing in a function (from the page that uses this component) when toggling the task as start/end. From there, a re-render (by doing another fetch) should be done such that the TaskCard will now reflect with the new state of Start or End

Reply:
I honestly forgot that toggle should trigger an API call. We can remove this since any change in state should be reflected on the server and the API call will update the local state. In other words, `TaskCard` will not have to hold any state. We only need the `CalendarPage` or `TaskHistoryPage` to maintain the most up-to-date copy of the tasks.

Reply:
I've removed the code highlighted above. I will fix it back when we merge `develop` into "calendar page linking".

Reply:
Sounds good!